Add recipe for org-xlatex

ksqsf opened the pull request:
### Brief summary of what the package does

Org-xlatex creates instant previews for Org LaTeX formulae using MathJax in xwidget.

### Direct link to the package repository

https://github.com/ksqsf/org-xlatex

### Your association with the package

I'm the maintainer.

### Relevant communications with the upstream package maintainer

None needed.

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] I've used `M-x checkdoc` to check the package's documentation strings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Thanks for this.  Quick first pass --

`org-xlatex.el` with byte-compile using Emacs 28.1:
```
org-xlatex.el:261:6: Warning: the function `kill-xwidget' is not known to be
    defined.
org-xlatex.el:260:34: Warning: the function `xwidget-live-p' is not known to
    be defined.
org-xlatex.el:237:59: Warning: the function `xwidget-webkit-display-callback'
    is not known to be defined.
```

`org-xlatex.el` with package-lint 20230806.1331:
```
2 issues found:
380:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
382:1: warning: `with-eval-after-load' is for use in configurations, and should rarely be used in packages.
```
I think you can remove the `with-eval-after-load` -- it's safe to call `add-hook` on `tab-line-mode-hook` before `tab-line` is loaded in.  But I'd also recommend moving these forms inside the body of your minor mode.

`org-xlatex.el` with [melpazoid](https://github.com/riscy/melpazoid):
```
- org-xlatex.el#L274: Mixed tab/space indentation, consider `(setq indent-tabs-mode nil)`
- org-xlatex.el#L322: There is no need to quote lambdas (neither #' nor ')
- org-xlatex.el#L197: It's safer to sharp-quote function names; use `#'`
```

Package and license:
- Prefer equivalent recipe: `(org-xlatex :fetcher github :repo "ksqsf/org-xlatex" :files (:defaults "*.html"))

(I'd also avoid `*.html` if you just plan on packaging a single html file; you don't want to inadvertently be packaging up a bunch of web docs with this)

ksqsf commented:
Hi @riscy , thanks for the review!

1. AFAICT, Emacs 28 did not define those functions. Emacs 29 made some changes to the xwidget event system, so it is quite involved to extend the support to 28. Therefore, I decided to support only 29 and newer versions. `org-xlatex.el` compiles cleanly using 29.
2. I removed `with-eval-after-load` and moved the `add-hook` calls to `org-xlatex--setup`. Also added corresponding `remove-hook` calls to `org-xlatex--teardown` to reduce environment pollution.

Other comments have been addressed accordingly.

riscy commented:
Thank you - I also notice my sharp-quote hint for `org-xlatex--setup` is a false-positive, and I've made a note of that.  Merging!
